Capture carb updates for create, update, and delete operations

darinkrauss opened the pull request:
- https://tidepool.atlassian.net/browse/LOOP-1417
- Update to use latest RemoteDataService.uploadCarbData
- Support new CarbStoreError error code
- Various updates to support using Result for completion handler
- Update CarbStore simulated core data
- Update CarbStoreTests

This PR is based upon changes to LoopKit in PR https://github.com/tidepool-org/LoopKit/pull/217.

Note: The next few PRs will be against the primary feature branch darinkrauss/LOOP-1417-capture-carbohydrate-entries.v2.ALL, not dev. My intention is to break the large overall feature into smaller, more easily reviewable PRs, and then merge the primary feature branch into dev after a final "summary" PR. (Several of the smaller PRs cannot be merged directly into dev because they would partially break existing functionality without a full replacement.)

Reviewers requested: 2

nhamming committed:
correcting the bolus entry colour (#213)

Commit:
Capture carb updates for create, update, and delete operations

- https://tidepool.atlassian.net/browse/LOOP-1417
- Update to use latest RemoteDataService.uploadCarbData
- Support new CarbStoreError error code
- Various updates to support using Result for completion handler
- Update CarbStore simulated core data
- Update CarbStoreTests

novalegra committed:
LOOP-1686: add higher level tests to Loop algorithm (#200)

LOOP-1686: add higher level tests to Loop algorithm

Commit:
Minor updates to cleanup code

nhamming commented on Loop Status Extension/StatusViewController.swift:
```diff
@@ -92,11 +92,11 @@ class StatusViewController: UIViewController, NCWidgetProviding {
     let absorptionTimes = LoopSettings.defaultCarbAbsorptionTimes
     lazy var carbStore = CarbStore(healthStore: healthStore,
-                                   observeHealthKitSamplesFromOtherApps: FeatureFlags.observeHealthKitSamplesFromOtherApps,
+                                   observeHealthKitSamplesFromOtherApps: false,
```
[nit] should this be hardcoded?

darinkrauss replied:
Yes, only the CarbStore for the main app needs to observe HealthKit so doing it it the status extension or watch would just be duplicating work.

darinkrauss replied:
Unless you can think of a use case where the status extension would be running (and able to ingest HealthKit data) where the main app would not be able to do so. I can't think of one.

nhamming reviewed (approved):
LGTM

novalegra committed:
LOOP-1830: Update carb edit screen text (#217)

Commit:
New Settings: action sheets for adding new devices (#215)

* checkpoint

* Remove SettingsUICoordinator (not ready yet)

* Also change lookup

* More refactoring to handle a pump-as-a-CGM

* minor tweaks

* minor tweaks

* minor tweaks

* PR Feedback

* Move functions out of extension they don't belong in

* Proposal addressing PR feedback with idea of using a shim

novalegra committed:
LOOP-1627: Add button color to communicate state on watch temporary correction range adjustments (#216)

LOOP-1627: Add button color to communicate state on watch temporary correction range adjustments

novalegra committed:
Fix watch storyboard connection (#220)

Commit:
Merge branch 'dev' into darinkrauss/LOOP-1417-capture-carbohydrate-entries.v2.carb-updates

Commit:
LOOP-1717: Swap new SettingsView for SettingsTableViewController (part 1 of N) (#219)

* checkpoint

* Remove SettingsUICoordinator (not ready yet)

* Also change lookup

* More refactoring to handle a pump-as-a-CGM

* minor tweaks

* minor tweaks

* minor tweaks

* PR Feedback

* Move functions out of extension they don't belong in

* checkpoint: sort of working

* Hook up services menu section

* Proposal addressing PR feedback with idea of using a shim

* Make sure updating device button works

* Fix merge error

Commit:
Merge branch 'dev' into darinkrauss/LOOP-1417-capture-carbohydrate-entries.v2.carb-updates

# Conflicts:
#	Loop/View Controllers/StatusTableViewController.swift

ps2 reviewed (approved):
LGTM!
